plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/LayerManager.h

```diff
@@ -30,7 +30,19 @@
 
 #include <map>
 #include <set>
-#include <boost/thread.hpp>
+
+// On Windows, against certain versions of Maya and with strict compiler
+// settings on, we are getting warning-as-error problems with a couple
+// boost includes.  Disabling those warnings for the specific includes
+// for now instead of disabling the strict settings at a higher level.
+#ifdef _MSC_VER
+#pragma warning( push )
+#pragma warning( disable : 4002 )
+#endif
+  #include <boost/thread.hpp>
+#ifdef _MSC_VER
+#pragma warning( pop )
+#endif
```

Comment:
No real problem against doing this. If this happens on a frequent occurrence, then a macro such as:
```c++
#define PUSH_DISABLE_WARNING_MSVC4002 \
  __pragma(warning(push)) \
  __pragma(warning(disable:4002)) 

#define PUSH_DISABLE_WARNING_MSVC4275 \
  __pragma(warning(push)) \
  __pragma(warning(disable:4275)) 

#define POP_DISABLE_WARNINGS \
  __pragma(warning(pop)) 
```
Might be an idea, but no real issue with this as is.